lib/system.nim

```diff
@@ -1345,6 +1345,13 @@ proc delete*[T](x: var seq[T], i: Natural) {.noSideEffect.} =
   ## .. code-block:: Nim
   ##  var i = @[1, 2, 3, 4, 5]
   ##  i.delete(2) # => @[1, 2, 4, 5]
+  if i > high(x):
+    var e: ref IndexDefect
+    new(e)
+    e.msg = "out of bounds!"
+    raise e
+    # raiseIndexError2(i, high(x))
```

Comment:
Can someone help me out here? I am not sure how to properly import/include `raiseIndexError2` from `chcks`, because `chcks` needs `sysFatal` and `sysFatal` needs `&`.

Reply:
I would try it this way:

```nim

    # by moving the first iteration out of the loop, we gurantee
    # that invalid indexes are caught:
    movingCopy(x[i], x[if i+1 < xl: i+1 else: xl-1])
    for j in i+1..xl-2: movingCopy(x[j], x[j+1])
```

Reply:
This would not raise the `IndexDefect` in the generated javascript.
Or do you mean, putting the first iteration to line 1348?

Reply:
this works:
move this right below `export dollars`:
```nim
proc delete*[T](x: var seq[T], i: Natural) {.noSideEffect.} =
  ## Deletes the item at index `i` by moving all `x[i+1..]` items by one position.
  ##
  ## This is an `O(n)` operation.
  ##
  ## See also:
  ## * `del <#del,seq[T],Natural>`_ for O(1) operation
  ##
  ## .. code-block:: Nim
  ##  var i = @[1, 2, 3, 4, 5]
  ##  i.delete(2) # => @[1, 2, 4, 5]
  if i > high(x):
    # xxx this should call `raiseIndexError2(i, high(x))` after some refactoring
    raise (ref IndexDefect)(msg: "index out of bounds: '" & $i & "' < '" & $x.len & "' failed")

  template defaultImpl =
    let xl = x.len
    for j in i.int..xl-2: movingCopy(x[j], x[j+1])
    setLen(x, xl-1)

  when nimvm:
    defaultImpl()
  else:
    when defined(js):
      {.emit: "`x`.splice(`i`, 1);".}
    else:
      defaultImpl()
```

then change `.. code-block:: Nim` to runnableExamples and also add a doAssertRaises example showing a failure case

Reply:
That makes sense, done.
However, now there's this error: `Error: undeclared identifier: '$'`

Reply:
My bad, failed to correctly read your comment. fixed.